listenbrainz_spark/stats/user/utils.py

```diff
@@ -1,51 +1,31 @@
+from datetime import datetime
 import time
 
 from collections import defaultdict
 from listenbrainz_spark.stats import run_query
+from listenbrainz_spark.utils import get_listens
+from listenbrainz_spark.stats import adjust_months
+from listenbrainz_spark.path import LISTENBRAINZ_DATA_DIRECTORY
 
-def get_artists(table):
-    """
-    Get artist information (artist_name, artist_msid etc) for every user
-    ordered by listen count (number of times a user has listened to tracks
-    which belong to a particular artist).
 
-    Args:
-        table: name of the temporary table.
+def get_latest_listen_ts():
+    now = datetime.now()
+    while True:
+        try:
+            df = get_listens(now, now, LISTENBRAINZ_DATA_DIRECTORY)
+            break
+        except:
+            now = adjust_months(now, 1)
 
-    Returns:
-        artists: A dict of dicts which can be depicted as:
-                {
-                    'user1': [{
-                        'artist_name': str,
-                        'artist_msid': str,
-                        'artist_mbids': str,
-                        'listen_count': int
-                    }],
-                    'user2' : [{...}]
-                }
-    """
-    t0 = time.time()
-    result = run_query("""
-            SELECT user_name
-                 , artist_name
-                 , artist_msid
-                 , artist_mbids
-                 , count(artist_name) as cnt
-              FROM %s
-          GROUP BY user_name, artist_name, artist_msid, artist_mbids
-          ORDER BY cnt DESC
-        """ % (table))
+    df.createOrReplaceTempView('latest_listen_ts')
+    result = run_query("SELECT MAX(listened_at) as max_date FROM latest_listen_ts")
```

Comment:
maybe call it `max_timestamp` or `max_listened_at` instead, clearer and more accurate.